x-pack/plugins/apm/common/locator.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { LocatorDefinition } from 'src/plugins/share/common';
+import type { SerializableRecord } from '@kbn/utility-types';
+import * as t from 'io-ts';
+import { isRight } from 'fp-ts/Either';
+
+const serviceOverViewTab = t.keyof({
+  transactions: null,
+  metrics: null,
+  logs: null,
+});
+
+export const APM_APP_LOCATOR = 'APM_LOCATOR';
+
+type ServiceOverViewTab = t.TypeOf<typeof serviceOverViewTab>;
+
+export interface ServiceViewLocatorParams extends SerializableRecord {
+  serviceName?: string;
+  serviceOverViewActiveTab?: ServiceOverViewTab;
+}
+
+export type APMLocatorParams = ServiceViewLocatorParams;
+
+export class APMLocatorDefinition
+  implements LocatorDefinition<APMLocatorParams>
+{
+  public readonly id = APM_APP_LOCATOR;
+
+  public readonly getLocation = async ({
+    serviceName,
+    serviceOverViewActiveTab,
+  }: APMLocatorParams) => {
+    let path;
+    if (
+      serviceName &&
+      serviceOverViewActiveTab &&
+      isRight(serviceOverViewTab.decode(serviceOverViewActiveTab))
+    ) {
+      path = `/services/${encodeURIComponent(serviceName)}/${encodeURIComponent(
```

Comment:
Can't we use `apmRouter.link` here? e.g. https://github.com/elastic/kibana/blob/d101d08a0676107c61320faceface1bb3d5e01b4/x-pack/plugins/apm/public/components/app/service_map/popover/service_contents.tsx#L84